### ISSUE TITLE
New version of bakta v1.10 fails to run annotations

### ISSUE THREAD

**Issue by Dx-wmc** [oschwengers/bakta#346]
Hi, I upgraded to version 1.10 using conda (the version 1.9 environment cannot be upgraded directly). When I tried to run the genome that could be run successfully in the previous version, it got an error.

The command I used: `bakta K130.fasta --prefix K130 --output K130_bakta --min-contig-length 200 --locus-tag K130 -t 16 --keep-contig-headers --force`

```
/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/numpy/_core/getlimits.py:555: UserWarning: The value of the smallest subnormal for <class 'numpy.float32'> type is zero.
  setattr(self, word, getattr(machar, word).flat[0])
/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/numpy/_core/getlimits.py:89: UserWarning: The value of the smallest subnormal for <class 'numpy.float32'> type is zero.
  return self._float_to_str(self.smallest_subnormal)
/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/numpy/_core/getlimits.py:555: UserWarning: The value of the smallest subnormal for <class 'numpy.float64'> type is zero.
  setattr(self, word, getattr(machar, word).flat[0])
/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/numpy/_core/getlimits.py:89: UserWarning: The value of the smallest subnormal for <class 'numpy.float64'> type is zero.
  return self._float_to_str(self.smallest_subnormal)
Bakta v1.10.0
Options and arguments:
	input: /home/data/usr/dx/workdir/test/bakta/K130.fasta
	db: /home/data/public/database/bakta/db, version 5.0, full
	translation table: 11
	locus tag prefix: K130
	output: /home/data/usr/dx/workdir/test/bakta/K130_bakta
	force: True
	tmp directory: /tmp/tmp4ifivcfw
	keep/sequence headers: True
	prefix: K130
	threads: 16
	debug: True

Bakta runs in DEBUG mode! Temporary data will not be destroyed at: /tmp/tmp4ifivcfw

Parse genome sequences...
	imported: 210
	filtered & revised: 210
	contigs: 210

Start annotation...
predict tRNAs...
	found: 83
predict tmRNAs...
	found: 1
predict rRNAs...
	found: 13
predict ncRNAs...
	found: 93
predict ncRNA regions...
	found: 57
predict CRISPR arrays...
	found: 0
predict & annotate CDSs...
	predicted: 5682 
	discarded spurious: 6
	revised translational exceptions: 3
	detected IPSs: 5515
	found PSCs: 128
	found PSCCs: 16
	lookup annotations...
	conduct expert systems...
		amrfinder: 31
		protein sequences: 244
	combine annotations and mark hypotheticals...
	detect pseudogenes...
		candidates: 9
		verified: 1
	analyze hypothetical proteins: 39
		detected Pfam hits: 1 
		calculated proteins statistics
	revise special cases...
detect & annotate sORF...
	detected: 57570
	discarded due to overlaps: 44771
	discarded spurious: 0
	detected IPSs: 34
	found PSCs: 1
	lookup annotations...
	filter and combine annotations...
	filtered sORFs: 28
detect gaps...
	found: 0
detect oriCs/oriVs...
	found: 6
detect oriTs...
	found: 2
apply feature overlap filters...
select features and create locus tags...
	selected: 5954
improve annotations...
	revised gene symbols: 33

Genome statistics:
	Genome size: 5,923,825 bp
	Contigs/replicons: 210
	GC: 56.8 %
	N50: 121,978
	N ratio: 0.0 %
	coding density: 87.7 %

annotation summary:
	tRNAs: 82
	tmRNAs: 1
	rRNAs: 13
	ncRNAs: 93
	ncRNA regions: 57
	CRISPR arrays: 0
	CDSs: 5674
		hypotheticals: 38
		pseudogenes: 1
	sORFs: 26
	gaps: 0
	oriCs/oriVs: 6
	oriTs: 2

Export annotation results to: /home/data/usr/dx/workdir/test/bakta/K130_bakta
	human readable TSV...
	GFF3...
	INSDC GenBank & EMBL...
Traceback (most recent call last):
  File "/home/data/usr/dx/miniconda3/envs/bakta/bin/bakta", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/bakta/main.py", line 543, in main
    insdc.write_features(data, features, genbank_path, embl_path)
  File "/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/bakta/io/insdc.py", line 292, in write_features
    sequence_list = build_biopython_sequence_list(data, features)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/bakta/io/insdc.py", line 248, in build_biopython_sequence_list
    qualifiers['note'].append("(3' truncated)")
    ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'append'
```

In addition, when running only `bakta -h`, there will still be warning messages.
```
/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/numpy/_core/getlimits.py:555: UserWarning: The value of the smallest subnormal for <class 'numpy.float32'> type is zero.
  setattr(self, word, getattr(machar, word).flat[0])
/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/numpy/_core/getlimits.py:89: UserWarning: The value of the smallest subnormal for <class 'numpy.float32'> type is zero.
  return self._float_to_str(self.smallest_subnormal)
/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/numpy/_core/getlimits.py:555: UserWarning: The value of the smallest subnormal for <class 'numpy.float64'> type is zero.
  setattr(self, word, getattr(machar, word).flat[0])
/home/data/usr/dx/miniconda3/envs/bakta/lib/python3.11/site-packages/numpy/_core/getlimits.py:89: UserWarning: The value of the smallest subnormal for <class 'numpy.float64'> type is zero.
  return self._float_to_str(self.smallest_subnormal)
```

The fasta file and the log file I have attached below.
[fasta+log.zip](https://github.com/user-attachments/files/17766129/fasta%2Blog.zip)

**Comment by mjfos2r** [comment]
I am also getting the NumPy warning. It appears to be related to some dependency somewhere that's [compiled with -ffast-math](https://moyix.blogspot.com/2022/09/someones-been-messing-with-my-subnormals.html) but I've no clue which it could be.

Annotation is working for me despite this.

**Comment by oschwengers** [comment]
Hi @Dx-wmc and @mjfos2r,
thanks for reporting. There are two distinct issues occurring here:

1. There is a true bug that caused the above reported crash. This is fixed by https://github.com/oschwengers/bakta/commit/5e7f09f366a1d75169a9634f5785439318acbffd and https://github.com/oschwengers/bakta/commit/6332cfdeb521a74af01f8ab8ba320ab97963f9bb
2. Though no real bugs, there are some very annoying warnings which are caused by ffast-match compilation flag as pointed out by @mjfos2r . Unfortunately, I also have no clue what exactly is causing this. I can only hope that someone has a clue or over time, this will be fixed upstream. 

Together with the fix for #342, I'll release a patch very soon. To separate reports, I've filed a dedicated issue for the warning messages and close this. For everything related to the warnings, please feel free to jump over to #347.
